pkg/cli/cli.go

```diff
@@ -983,23 +1001,165 @@ func printConnection(c *command, cvd RemoteCVDLocator, status ConnStatus) {
+	host, err := findHost(service, flags.host)
+	if err != nil {
+		return fmt.Errorf("failed to find host")
+	}
+	if host.Docker == nil {
+		return errors.New("instance type should be Docker")
+	}
+
+	// Retriving ADB port of Cuttlefish device
+	controlDir := opts.InitialConfig.ConnectionControlDirExpanded()
+	cvd, err := findCVD(service, controlDir, flags.host, device)
+	if err != nil {
+		return fmt.Errorf("failed to find cvd: %w", err)
+	}
+	_, port, err := net.SplitHostPort(cvd.ADBSerial)
+	if err != nil {
+		return fmt.Errorf("failed to parse port from ADB serial: %w", err)
+	}
+	if port == "" {
+		return errors.New("failed to find port")
+	}
+
+	// Make connection towards [host_ip_address]:[cuttlefish_instance_adb_port]
+	var dialer proxy.Dialer
+	if flags.HTTPProxy != "" {
+		proxyUrl, err := url.Parse(flags.HTTPProxy)
+		if err != nil {
+			return fmt.Errorf("failed to parse proxy URL: %w", err)
+		}
+		if proxyUrl.Scheme != "socks5" {
+			return fmt.Errorf("scheme of proxy URL is not socks5. actual: %s", proxyUrl.Scheme)
+		}
+		dialer, err = proxy.SOCKS5("tcp", proxyUrl.Host, nil, nil)
+		if err != nil {
+			return fmt.Errorf("failed to create proxy dialer: %w", err)
+		}
+	} else {
+		dialer = proxy.Direct
+	}
+	remoteConn, err := dialer.Dial("tcp", net.JoinHostPort(host.Docker.IPAddress, port))
+	if err != nil {
+		return fmt.Errorf("failed to dial remote port: %w", err)
+	}
+	defer remoteConn.Close()
+
+	// Create a file socket to establish ADB connection
+	socketPath := GetProxySocketPath(controlDir, flags.host, device)
+	localListener, err := net.Listen("unix", socketPath)
+	if err != nil {
+		return fmt.Errorf("failed to create unix socket: %w", err)
+	}
+	defer localListener.Close()
+
+	// Enroll signal channel to handle SIGINT and SIGTERM.
+	sigCh := make(chan os.Signal, 1)
+	signal.Notify(sigCh, syscall.SIGINT, syscall.SIGTERM)
+
+	// Initialize ADB connection
+	errorCh := make(chan error)
+	go func() {
+		if err := enableAdbConnection(socketPath, true); err != nil {
+			errorCh <- fmt.Errorf("failed to connect adb: %w", err)
+		}
+	}()
+
+	// Listener is accepted by ADB connection.
+	startLocalToRemoteCh := make(chan net.Conn)
+	startRemoteToLocalCh := make(chan net.Conn)
+	go func() {
+		localConn, err := localListener.Accept()
+		if err != nil {
+			errorCh <- fmt.Errorf("failed to accept connection: %w", err)
+
+		} else {
+			startLocalToRemoteCh <- localConn
+			startRemoteToLocalCh <- localConn
+		}
+	}()
+
+	// After acception, start data transfer to do the role proxying ADB connection.
+	go func() {
+		localConn := <-startLocalToRemoteCh
+		if _, err := io.Copy(remoteConn, localConn); err != nil {
+			errorCh <- fmt.Errorf("no longer able to copy data from local to remote: %w", err)
+		}
+	}()
+	go func() {
+		localConn := <-startRemoteToLocalCh
+		if _, err := io.Copy(localConn, remoteConn); err != nil {
+			errorCh <- fmt.Errorf("no longer able to copy data from remote to local: %w", err)
+		}
+	}()
```

Comment:
it would be another separate method as well(forward?)

Reply:
Done.